code-generation/generator/src/main/resources/com/amazonaws/util/awsclientgenerator/velocity/cpp/s3/S3ClientHeader.vm

```diff
@@ -204,6 +203,7 @@ namespace ${rootNamespace}
 
 #end
     private:
+        friend class Aws::Client::ClientWithAsyncTemplateMethods<${className}>;
```

Comment:
"Let's be friends, but I won't make it public"
Joking, just did not want to pollute the "header" of the client header.